src/zjs_util.h

```diff
@@ -275,4 +275,66 @@ int zjs_validate_args(const char *expectations[], const jerry_length_t argc,
         (zjs_validate_args((const char *[]){ __VA_ARGS__, NULL }, argc, argv) \
                 <= ZJS_INVALID_ARG) ? 1 : 0
 
+#define ZJS_FIND_NODE(type, list, cmp_element, cmp_to) \
+   ({ \
+       type *found = NULL; \
+       type *cur = list; \
+       while (cur) { \
+           if (cur->cmp_element == cmp_to) { \
+               found = cur; \
+               break; \
+           } \
+           cur = cur->next; \
+       } \
+       found; \
+   })
+
+#define ZJS_APPEND_NODE(type, list, p) \
+    { \
+        type **pnext = &list; \
+        while (*pnext != NULL) { \
+            pnext = &(*pnext)->next; \
+        } \
+    *pnext = p; \
+    }
+#define ZJS_PREPEND_NODE(type, list, p) \
+    { \
+        ((type *)p)->next = list; \
```

Comment:
My reasoning for this was kinda dumb, but for the unit test code I just create list items on the stack, so I am passing in &item, which wont work, since you cant do &item->next. This is why I did the casting. Probably better just to do a malloc in the unit tests code, OR just create a pointer that points to &item and pass that.